src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -37,10 +41,12 @@
 (defn- build-rule
   "Build a new Rule based on the provided config-map"
   [config-map]
-  (let [rule-type (keyword (get config-map :type :path))]
-    (if (= rule-type :path)
-      (-> (rules/new-path-rule (config-map :path) (method config-map)))
-      (-> (rules/new-regex-rule (config-map :path) (method config-map))))))
+  (let [rule-type (keyword (get config-map :type :path))
+        rule-fn (rule-type new-rule-func-map)
+        rule (rule-fn (config-map :path) (method config-map))]
+    (if (true? (:allow_unauthenticated config-map))
+      (assoc rule :allow_unauthenticated true)
```

Comment:
What if we had the `config->rule` always insert optional keys with their default values?  Presumably we don't have any situations where the lack of a key in the configuration means something different than it's default value.

I did this with the query-param support - the configuration file can have a single string or a list of strings, but once it's parsed by `config->rule` the in-memory rule will always have a list of strings (which may contain a single string).

I think this sort of general pattern might help in reducing this sort of conditional checking.  This part could be written as:
```clj
(assoc rule :allow_unauthenticated (:allow_unauthenticated config-map)
```
...or a `->>`, whatever.

Granted, I haven't traced this all the way through the code to see if there'd be a problem with this approach in this case, but it's a valid consideration nonetheless.

@jeffmccune Would that be an improvement?